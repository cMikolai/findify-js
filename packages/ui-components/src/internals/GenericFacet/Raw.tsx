import * as React from 'react';
import * as cx from 'classnames';

const styles = require('./styles.css');

export interface Props {
  isMobile: boolean;
  children?: React.ReactChild;
}

export const Raw = ({ children, ...rest }: Props) => (
  <div className={cx(styles.content, rest.isMobile && styles.mobile)}>
    {React.Children.map(children, (child: React.ReactElement<any>) =>
      React.cloneElement(child, rest)
    )}
  </div>
);
