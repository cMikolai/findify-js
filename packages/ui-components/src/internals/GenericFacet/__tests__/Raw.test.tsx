import * as React from 'react';
import { Raw, Props } from '../Raw';

const Dummy = ({ text }: { text: string }) => <div>{text}</div>;

describe('<Raw />', () => {
  it('renders correctly', () => {
    const properties: Props[] = [
      { isMobile: true },
      { isMobile: false },
      { isMobile: true, children: <Dummy text="foo" /> },
      { isMobile: false, children: <Dummy text="bar" /> },
    ];
    properties.forEach((props: Props) => {
      const component = render(<Raw {...props} />);
      expect(component).toMatchSnapshot();
    });
  });
});
