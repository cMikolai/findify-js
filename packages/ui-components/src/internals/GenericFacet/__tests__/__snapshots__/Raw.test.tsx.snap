// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Raw /> renders correctly 1`] = `
<div
  class="content mobile"
/>
`;

exports[`<Raw /> renders correctly 2`] = `
<div
  class="content"
/>
`;

exports[`<Raw /> renders correctly 3`] = `
<div
  class="content mobile"
>
  <div>
    foo
  </div>
</div>
`;

exports[`<Raw /> renders correctly 4`] = `
<div
  class="content"
>
  <div>
    bar
  </div>
</div>
`;
